### PR TITLE
Href refactor: macros and playerpanel

### PR DESCRIPTION
**Pull request by MarsM0nd**
Only effective change should be that href links don't point at new objects because the old one was removed and a new one was added, stopping unintended actions.
Otherwise it should make the code more readable thanks to the macros.
:cl:
tweak:Admin links now should not point at the wrong person, but rather fail to have an effect.
/:cl:

### CODE REVIEW

**Comment by Kyep** [comment]
I'll be so happy when messages sent to the wrong person due to /ref issues are a thing of the past.

**Comment by MarsM0nd** [comment]
There should be only UID usage now for the hrefs: `adminmoreinfo, adminplayerobservefollow, adminplayeropts, vars, subtlemessage, traitor, BlueSpaceArtillery, CentcommReply, SyndicateReply, adminspawncookie, priv_msg, revive, follow`. Currently, they will still be worked on by locate. 
The list is mostly to allow others to see if they think a href would be missing, and for me to check for them that locateUID does get used on it.
As a small piece of trivia, the ammount of \ref found in the code went down from 539 to 440.

**Comment by MarsM0nd** [comment]
As the commit states, made sure that `code\modules\admin\topic.dm` does use locateUID on hrefs, except for 2 cases: Adminnewscaster, which shouldn't change around that a rogue \ref can happen, and some fax hrefs, which can't simply be turned into UID(), as the thing the \ref points at is saved as untyped var, which doesn't know UID().
Overall \ref count down to 341.

**Comment by SamHPurp** [comment]
Will be super happy to get rid of the last of the `\ref` ... this is one of the things that conflicts with `512`....

**Comment by MarsM0nd** [comment]
Reduced the scope of this PR, should cover most of the admin tools. There should be some understanding on how a specific subsystem uses the references, and i don't have the time to, nor should the already made changes wait for me to, work through all the systems that still use \ref. It also allows other to take a system and work on it without thinking it will colide with this PR, or another, if it gets focused on a specific system. To name a few, mechs, the event manager, and pAI recruitment still use \refs.

**Comment by variableundefined** [comment]
This conflict with the current set of TM on the server. I'll inform you when I am ready to TM it.

File: code/defines/procs/admin.dm [hunk 1]
```diff
@@ -68,7 +68,10 @@
 	return .
 
 /proc/key_name_admin(whom)
-	var/message = "[key_name(whom, 1)](<A HREF='?_src_=holder;adminmoreinfo=\ref[whom]'>?</A>)[isAntag(whom) ? "<font color='red'>(A)</font>" : ""][isLivingSSD(whom) ? "<span class='danger'>(SSD!)</span>" : ""] ([admin_jump_link(whom)])"
+	var/mob/whom_mob = whom //Should be always be a mob.
+	if(!istype(whom_mob)) //Best to check anyway
```
Comment (variableundefined) [inline code/defines/procs/admin.dm]: I don't think the istype check is working properly (There's no typepath to check again), it is returning false positive on live server.

File: code/defines/procs/admin.dm [hunk 2]
```diff
@@ -68,7 +68,10 @@
...
+		return  "key_name_admin did get a non-mob as argument. Argument is [whom]. Report it to a coder."
```
Comment (variableundefined) [inline code/defines/procs/admin.dm]: Can you change the phrasing to "Make an issue report or report it to a coder" - I am sick of people constantly pinging me and coders for this.

**Comment by variableundefined** [comment]
@MarsM0nd PR is being test merged and some issues were foudn

**Comment by MarsM0nd** [comment]
Merge conflict fixed and the problematic check turned into a check for null. Should catch the only case where it runtimes (trying to run UID from a null object), typecast is still needed as the compiler otherwise doesn't know the object has the UID proc.

File: code/defines/procs/admin.dm [hunk 3]
```diff
@@ -68,8 +68,10 @@
 	return .
 
 /proc/key_name_admin(whom)
-	var/message = "[key_name(whom, 1)](<A HREF='?_src_=holder;adminmoreinfo=\ref[whom]'>?</A>)[isAntag(whom) ? "<font color='red'>(A)</font>" : ""][isLivingSSD(whom) ? "<span class='danger'>(SSD!)</span>" : ""] ([admin_jump_link(whom)])"
-	return message
+	if(whom)
+		var/mob/whom_mob = whom //As long as it's not null, will be close enough/has the proc UID() that is all that's needed
```
Comment (tigercat2000) [inline code/defines/procs/admin.dm]: Cast this as whatever the highest level type UID is defined on is. I think that's either atom or datum.

**Comment by variableundefined** [comment]
@MarsM0nd Looking good, just need to address Tiger's concern.

**Pull request merged**